src/backend/distributed/shared_library_init.c

```diff
@@ -678,6 +678,9 @@ StartupCitusBackend(void)
 	 * the gpid from the application_name.
 	 */
 	AssignGlobalPID();
+
+	SetBackendDataAfterAuthData();
+
```

Comment:
it's a bit strange to me that we do so much stuff in this function which can be called arbitrarily many times (e.g. if the Citus extension does not exist)

any alternative place for the last 4 calls in this function?